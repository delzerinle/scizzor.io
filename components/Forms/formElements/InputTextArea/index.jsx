import PropTypes from 'prop-types';
import { Field, ErrorMessage } from 'formik';

const InputTextArea = ({ name, rows = 10, label, errors, touched }) => {
  return (
    <div
      className={`border w-full border-alt-2 field-input ${
        errors?.[name] && touched?.[name] ? 'mb-10 field-error' : 'mb-4'
      }`}
    >
      <div className="px-4 py-2 relative">
        <Field
          id={name}
          name={name}
          rows={rows}
          component="textarea"
          className="w-full  field-textarea__field focus:outline-none relative z-10 bg-transparent font-light text-sm"
        />
        <label
          className="text-secondary font-light tracking-normal absolute px-4 left-0 transition-transform duration-200 ease-in z-0 field-textarea__label text-sm pt-1"
          htmlFor={name}
        >
          {label}
        </label>
        <ErrorMessage
          name={name}
          component="p"
          className="p-sm absolute text-secondary"
        />
      </div>
    </div>
  );
};

InputTextArea.propTypes = {
  rows: PropTypes.number,
  errors: PropTypes.object,
  touched: PropTypes.object,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default InputTextArea;
