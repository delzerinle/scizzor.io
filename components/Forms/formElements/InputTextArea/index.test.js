import React from 'react';
import { shallow } from 'enzyme';

import InputTextArea from './index';

describe('InputTextArea tests', () => {
  it('renders without crashing', () => {
    shallow(<InputTextArea label="Test" name="test" />);
  });

  it('input field snapshot', () => {
    const wrapper = shallow(
      <InputTextArea
        name="test"
        label="test"
        errors={{ test: 'test' }}
        touched={{ test: 'test' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
