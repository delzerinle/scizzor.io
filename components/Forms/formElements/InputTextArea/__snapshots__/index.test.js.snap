// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextArea tests input field snapshot 1`] = `
<div
  className="border w-full border-alt-2 field-input mb-10 field-error"
>
  <div
    className="px-4 py-2 relative"
  >
    <Field
      className="w-full  field-textarea__field focus:outline-none relative z-10 bg-transparent font-light text-sm"
      component="textarea"
      id="test"
      name="test"
      rows={10}
    />
    <label
      className="text-secondary font-light tracking-normal absolute px-4 left-0 transition-transform duration-200 ease-in z-0 field-textarea__label text-sm pt-1"
      htmlFor="test"
    >
      test
    </label>
    <FormikConnect(ErrorMessageImpl)
      className="p-sm absolute text-secondary"
      component="p"
      name="test"
    />
  </div>
</div>
`;
